### PR TITLE
Support CrowdPose training

### PR DESCRIPTION
**Pull request by bowenc0221**
1. Support training on the CrowdPose dataset
2. Support the latest PyTorch version (v1.4.0)
3. Support training with SyncBN

### CODE REVIEW

File: lib/dataset/CrowdPoseKeypoints.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        #     if obj['iscrowd']:
+        #         rle = crowdposetools.mask.frPyObjects(
+        #             obj['segmentation'], img_info['height'], img_info['width'])
+        #         m += crowdposetools.mask.decode(rle)
+        #     elif obj['num_keypoints'] == 0:
+        #         rles = crowdposetools.mask.frPyObjects(
+        #             obj['segmentation'], img_info['height'], img_info['width'])
+        #         for rle in rles:
+        #             m += crowdposetools.mask.decode(rle)
+
```
Comment (leoxiaobin) [inline lib/dataset/CrowdPoseKeypoints.py]: direct delete the commented code.

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 EasyDict==1.7
-opencv-python==3.4.1.15
+opencv-python
```
Comment (leoxiaobin) [inline requirements.txt]: why delete the version?

File: tools/crowdpose_concat_train_val.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# ------------------------------------------------------------------------------
```
Comment (leoxiaobin) [inline tools/crowdpose_concat_train_val.py]: why add '_concat_' in the file name?

File: tools/dist_train.py [hunk 4]
```diff
@@ -271,12 +277,13 @@ def main_worker(
         )
 
     for epoch in range(begin_epoch, cfg.TRAIN.END_EPOCH):
-        lr_scheduler.step()
-
         # train one epoch
         do_train(cfg, model, train_loader, loss_factory, optimizer, epoch,
                  final_output_dir, tb_log_dir, writer_dict, fp16=cfg.FP16.ENABLED)
 
+        # In PyTorch 1.1.0 and later, you should call `lr_scheduler.step()` after `optimizer.step()`.
```
Comment (leoxiaobin) [inline tools/dist_train.py]: We can consider only support PyTorch >= 1.1.0.

File: requirements.txt (hunk 2 above)
Comment (bowenc0221) [reply requirements.txt]: I could not find this version of opencv-python on my local machine using conda python.

File: tools/crowdpose_concat_train_val.py (hunk 3 above)
Comment (bowenc0221) [reply tools/crowdpose_concat_train_val.py]: This is used to concat train and val data to a single file.

File: tools/dist_train.py (hunk 4 above)
Comment (bowenc0221) [reply tools/dist_train.py]: Done, I have updated README accordingly.

File: lib/dataset/CrowdPoseKeypoints.py (hunk 1 above)
Comment (bowenc0221) [reply lib/dataset/CrowdPoseKeypoints.py]: Done, I have deleted these comments.

**Pull request merged**